### PR TITLE
Authorize resource for supervisor

### PR DESCRIPTION
**Pull request by trinhducduy**

### CODE REVIEW

File: app/models/course.rb [hunk 1]
```diff
@@ -40,4 +40,8 @@ def finished? user
     user_tasks = self.user_tasks.filter_by_user user
     user_tasks.count == self.tasks.count
   end
+
+  def owned_by? user
+    !course_users.find_by(user_id: user.id, is_owner: true).nil?
```
Comment (ThuBM) [inline app/models/course.rb]: use detect

File: app/views/supervisor/courses/_course.html.erb [hunk 2]
```diff
@@ -12,6 +12,24 @@
...
+        <%= t "supervisor.courses.general.owner" %>
+      </span>
+    <% else %>
+      <span class="label label-default">
+        <%= t "supervisor.courses.general.supervisor" %>
+      </span>
+    <% end %>
+  </td>
+  <td>
+    <% if course.is_active %>
```
Comment (ThuBM) [inline app/views/supervisor/courses/_course.html.erb]: change name to active?

Comment (trinhducduy) [inline app/views/supervisor/courses/_course.html.erb]: This is an attribute not function :-?

Comment (trinhducduy) [inline app/views/supervisor/courses/_course.html.erb]: so change attribute to active and use active?

File: app/models/course.rb [hunk 3]
```diff
@@ -40,4 +40,9 @@ def finished? user
     user_tasks = self.user_tasks.filter_by_user user
     user_tasks.count == self.tasks.count
   end
+
+  def owned_by? user
+    !course_users.detect{|course_user| course_user.user_id == user.id &&
+      course_user.is_owner == true}.nil?
```
Comment (ThuBM) [inline app/models/course.rb]: remove .nil?

**Comment by ThuBM** [comment]
OK

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -8,7 +8,11 @@ def create
       log_in user
       params[:session][:remember] == "1" ? remember(user) : forget(user)
       flash[:success] = t "application.flash.login_success"
-      redirect_to root_url
+      if user.supervisor?
```
Comment (dieunb) [inline app/controllers/sessions_controller.rb]: one line of code

**Pull request merged**